src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -1,65 +1,162 @@
 
-const ITEM_HEIGHT = 44;
+// https://github.com/atlassian/react-beautiful-dnd/issues/128#issuecomment-669083882
+function useDraggableInPortal() {
+  const self = useRef({}).current;
 
-export default function Sidebar({section, onChange, categories}) {
+  useEffect(() => {
+    const div = document.createElement('div');
+    div.classList.add(styles.portal);
+    self.elt = div;
+    document.body.appendChild(div);
+    return () => {
+      document.body.removeChild(div);
+    };
+  }, [self]);
+
+  return (render) =>
+    (provided, ...args) => {
+      const element = render(provided, ...args);
+      if (provided.draggableProps.style.position === 'fixed') {
+        return createPortal(element, self.elt);
+      }
+      return element;
+    };
+}
+
+export default function Sidebar({section, onClick, categories: initialCategories}) {
   const containerRef = useRef(null);
+  const [categories, setCategories] = useState(initialCategories);
 
-  useEffect(() => {
-    const currentRef = containerRef.current;
-    if (section.eventKey == null || currentRef == null) return;
+  const renderDraggable = useDraggableInPortal();
+  useEffect(() => setCategories(initialCategories), [initialCategories]);
 
-    const top = currentRef.scrollTop;
-    const index = categories.findIndex((category) => category.id === section.eventKey);
-    const depth = index * ITEM_HEIGHT;
+  useAutoScroll(section, containerRef, categories);
 
-    let newTop;
-    if (depth < top) {
-      newTop = depth;
-    }
-    if (depth + ITEM_HEIGHT > top + WindowHeight) {
-      newTop = depth - WindowHeight + ITEM_HEIGHT;
-    }
-    if (newTop == null) {
-      return;
+  const [hovering, setHovering] = useState(false);
+
+  const [topCategories, middleCategories, bottomCategories] = useMemo(() => {
+    const top = [];
+    const middle = [];
+    const bottom = [];
+
+    for (const category of categories) {
+      if (TOP_FIXED_CATEGORIES.includes(category.id)) {
+        top.push(category);
+        continue;
+      }
+
+      if (BOTTOM_FIXED_CATEGORIES.includes(category.id)) {
+        bottom.push(category);
+        continue;
+      }
+
+      middle.push(category);
     }
 
-    currentRef.scrollTo({
-      top: newTop,
+    return [top, middle, bottom];
+  }, [categories]);
+
+  const handleEmojiClick = useCallback(() => {
+    containerRef.current.scrollTo({
+      top: (topCategories.length + middleCategories.length) * ITEM_HEIGHT,
       left: 0,
-      behavior: 'smooth',
     });
-  }, [section]);
+
+    onClick(bottomCategories[0].id);
+  }, [containerRef, topCategories, middleCategories, bottomCategories]);
+
+  const handleReorder = useCallback(
+    (oldDest, newDest) => {
+      if (oldDest === newDest) {
+        return;
+      }
+
+      const result = [...middleCategories];
+      const [removed] = result.splice(oldDest, 1);
+      result.splice(newDest, 0, removed);
+
+      setCategories([...topCategories, ...result, ...bottomCategories]);
```

Comment:
it isnt clear why we have stored top and bottom categories in this when they are immutable